numpy/core/src/multiarray/arrayobject.c

```diff
@@ -703,10 +703,13 @@ PyArray_FailUnlessWriteable(PyArrayObject *obj, const char *name)
    If they are NULL terminated, then stop comparison.
 */
 static int
-_myunincmp(npy_ucs4 *s1, npy_ucs4 *s2, int len1, int len2)
+_myunincmp(npy_ucs4 const *s1, npy_ucs4 const *s2, int len1, int len2)
 {
-    npy_ucs4 *sptr;
-    npy_ucs4 *s1t=s1, *s2t=s2;
+    npy_ucs4 const *sptr;
+    /* FIXME: Casting away const makes the below easier to write, but should
+     * still be safe.
+     */
+    npy_ucs4 *s1t = (npy_ucs4 *)s1, *s2t = (npy_ucs4 *)s2;
```

Comment:
Same as above, didn't want to hide a rewrite amongst a patch that introduces `const` repeatedly and nothing else